docs/contracts/v4/concepts/01-v4-vs-v3.mdx

```diff
@@ -1,30 +1,68 @@
 ---
-title: v4-versus-v3
+title: V4 vs V3
 ---
 
-While Uniswap v4's underlying concetrated liquidity is the same as Uniswap v3, there are some key differences in the architecture
-and accounting.
+While Uniswap v4's underlying concentrated liquidity is the same as Uniswap v3,
+there are some key differences in the architecture and accounting.
 
 # Singleton Design
 
 ### Pool Creation
 
-(TODO: talk about singleton vs factory, state vs contract/V3Pool)
+V4: The singleton contract facilitates the creation of a pool and 
+also stores its state. This pattern reduces costs when creating a pool
+and doing multi-hop swaps.
+
+V3: A factory contract is responsible for pool creation. The pool is
+a separate contract instance that manages its own state. Pool initialization
+is costly because of how gas consuming contract creation is.
 
 ### Flash Accounting
 
-(TODO: talk about flash accounting, multi-hop swaps)
+V4: The singleton uses "flash accounting", meaning a caller that unlocks it
+is allowed to freely access tokens given
+that prior to locking it back no tokens are owed to or from the caller. This
+further reduces the gas cost of
+trades that cross multiple pools and supports more complex
+integrations with Uniswap v4.
+
+V3: Because flash accounting is missing from V3, it is the responsibility
+of the swap router to execute swaps on each individual pool from which the
+multi-hop swap needs access to liquidity.
```

Comment:
```suggestion
**V3**: Because flash accounting is missing from V3, it is the responsibility
of the integrating contract to perform token transfers, after each individual call, to each individual pool contract
```